grr/server/grr_response_server/gui/ui/components/flow_details/plugins/osquery_details_test.ts

```diff
@@ -1,111 +1,75 @@
-import {async, TestBed, ComponentFixture} from '@angular/core/testing';
-import {initTestEnvironment} from '@app/testing';
+import { async, TestBed, ComponentFixture} from '@angular/core/testing';
+import { initTestEnvironment } from '@app/testing';
 
-import {PluginsModule} from './module';
+import { PluginsModule } from './module';
 import { OsqueryDetails } from './osquery_details';
-import { newFlowListEntry, newFlowResultSet } from '@app/lib/models/model_test_util';
-import { FlowState, FlowListEntry, FlowResultSet } from '@app/lib/models/flow';
-import { By } from '@angular/platform-browser';
+import { FlowState, FlowListEntry } from '@app/lib/models/flow';
 import { DebugElement } from '@angular/core';
-import { OsqueryColumn, OsqueryRow } from '@app/lib/api/api_interfaces';
-import { DeepMutable } from '@app/lib/type_utils';
+import { By } from '@angular/platform-browser';
+import { newFlowListEntry, newFlowResultSet, newFlow } from '@app/lib/models/model_test_util';
 
 initTestEnvironment();
 
-/** Helper class to build a FlowListEntry objects in a declarative manner */
-class FlowListEntryBuilder {
-  private flowListEntry = newFlowListEntry({args: {query: ''}})  as DeepMutable<FlowListEntry>;
-
-  private stderr = '';
-  private table = {
-    query: '',
+/**
+ * Builds an OsqueryTable
+ * @param query The Osquery query which produced this table
+ * @param columns Column names of the table
+ * @param rows Array of arrays containing row values
+ */
+export function newOsqueryTable(
+  query: string,
+  columns: ReadonlyArray<string>,
+  rows: ReadonlyArray<ReadonlyArray<string>>,
+) {
```

Comment:
Probably makes sense to specify the return type here, so that the type checker can catch the errors early.

Reply:
Another note - if a test function is shared between two tests, let's put it in a third file. So maybe, for this purpose, we'd need to resurrect the osquery_test_utils.ts that was in this PR earlier. Primary reason: more rigid build systems, like Bazel, are sometimes not happy when a single source file has a dual purpose (a test and a library file with exported functions).

Alternatively, we can put this function into https://github.com/google/grr/blob/master/grr/server/grr_response_server/gui/ui/lib/models/model_test_util.ts